editoast/openapi.yaml

```
@@ -4132,4 +4123,42 @@ components:
         - slopes
         - curves
         - base
-        - electrification_conditions
+        - electrification_ranges
+        - power_restriction_ranges
+
+    ElectrificationRange:
+      properties:
+        start:
+          type: number
+          format: double
+        stop:
+          type: number
+          format: double
+        electrificationUsage:
+          oneOf:
+            - $ref: "#/components/schemas/Electrified"
+            - $ref: "#/components/schemas/Neutral"
+            - $ref: "#/components/schemas/NonElectrified"
+          discriminator:
+            propertyName: object_type
+      required: [start, stop, is_electrified, electrification]
+
+    Electrified:
+      properties:
+        mode:
+          type: string
+        mode_handled:
+          type: boolean
+        profile:
+          type: string
+          nullable: true
+        profile_handled:
+          type: boolean
+      required: [mode, mode_handled, profile_handled]
```

discriminator `object_type` must be on all the elements of the union

```suggestion
    Electrified:
      properties:
        mode:
          type: string
        mode_handled:
          type: boolean
        profile:
          type: string
          nullable: true
        profile_handled:
          type: boolean
        object_type:
          type: string
          enum: ["electrified"]
      required: [mode, mode_handled, profile_handled]
```

---

done